frontend/src/Components/Form/NumberInput.js

```diff
@@ -10,7 +10,7 @@ function parseValue(props, value) {
   } = props;
 
   if (value == null || value === '') {
-    return min;
+    return null;
```

Comment:
This ensures that a '' or null value in the input doesn't get changed to min value on a click in and out.